hibernate-osgi/src/main/java/org/hibernate/osgi/OsgiClassLoader.java

```diff
@@ -51,32 +60,37 @@ public OsgiClassLoader() {
 	@Override
 	@SuppressWarnings("rawtypes")
 	protected Class<?> findClass(String name) throws ClassNotFoundException {
-		if ( classCache.containsKey( name ) ) {
-			return classCache.get( name );
-		}
-		
-		for ( Bundle bundle : bundles ) {
-			try {
-				final Class clazz = bundle.loadClass( name );
-				if ( clazz != null ) {
-					classCache.put( name, clazz );
-					return clazz;
-				}
-			}
-			catch ( Exception ignore ) {
-			}
+	    Class< ? > cachedClass = classCache.get( name );
+	    if ( cachedClass != null ) {
+			return cachedClass;
 		}
-		
-		for ( ClassLoader classLoader : classLoaders ) {
-			try {
-				final Class clazz = classLoader.loadClass( name );
-				if ( clazz != null ) {
-					classCache.put( name, clazz );
-					return clazz;
-				}
-			}
-			catch ( Exception ignore ) {
-			}
+
+	    synchronized (bundles) {
```

Comment:
If we use `synchronized` every time we access `bundles`, then `bundles` can be a `HashSet` instead of a `Collections.synchronized` wrapper

Reply:
@vladmihalcea Hi - I left it intentionally a LinkedHashSet, since this was implemented before with HHH-12553 to retain the insertion-order. Using a HashSet would break this (and not be synchronized per-se as well, maybe you meant Hashmap?)

Reply:
Hey @vladmihalcea, @MrEasy is correct.  Super important that we maintain insertion order for these maps.

Reply:
@brmeyer thanks for reviewing